compiler/passes/codegen.cpp

```diff
@@ -360,6 +361,38 @@ compareSymbol(const void* v1, const void* v2) {
   return result;
 }
 
+static int
+compareSymbol2(void* v1, void* v2) {
+  Symbol* s1 = (Symbol*)v1;
+  Symbol* s2 = (Symbol*)v2;
+  ModuleSymbol* m1 = s1->getModule();
+  ModuleSymbol* m2 = s2->getModule();
+  if (m1 != m2) {
```

Comment:
I'm not a big fan of "if"; I try to use it as little as possible.  I'm even more prejudiced against "else".  So I would say something like:
  if (m1->modTag == m2->modTag) {
    // True if m1->cname strictly precedes m2->cname in lexical ordering.
    return strcmp(m1->cname, m2->cname) < 0;
  }
Note that strcmp does *not* promise to return -1 if the first argument sorts before the second, so exact comparison with -1 will fail on many interesting cases.